example/app/src/main.rs

```diff
@@ -0,0 +1,249 @@
+enum Command {
+    DeployContract,
+    ValidateSolution,
+}
+
+#[tokio::main]
+async fn main() {
+    let cli = Cli::parse();
+    let now = SystemTime::now();
+    env_logger::Builder::from_env(env_logger::Env::default().default_filter_or("info")).init();
+
+    let EnvironmentVariables {
+        rpc_url,
+        chain,
+        batcher_addr,
+        batcher_eth_addr,
+        eth_rpc_url,
+        proof_generator_addr,
+        keystore_path,
+        private_key,
+    } = EnvironmentVariables::new().unwrap_or_else(|err| {
+        error!("{}", err);
+        process::exit(1);
+    });
+
+    let wallet_alloy =
+        wallet_alloy::get_wallet(&chain, keystore_path.as_deref(), private_key.as_deref())
+            .unwrap_or_else(|err| {
+                error!("{}", err);
+                process::exit(1);
+            });
+
+    let provider = ProviderBuilder::new()
+        .with_recommended_fillers()
+        .wallet(wallet_alloy)
+        .on_http(
+            reqwest::Url::parse(&eth_rpc_url)
+                .map_err(|err| err.to_string())
+                .unwrap(),
+        );
+
+    match cli.command {
+        Command::DeployContract => {
+            // TODO(xqft): we might as well use the Chain type from Alloy, it isn't right to add
+            // aligned-sdk as a dependency only for this type.
+            let state_settlement_addr = match chain {
+                Chain::Devnet => BRIDGE_DEVNET_ETH_ADDR,
+                Chain::Holesky => BRIDGE_HOLESKY_ETH_ADDR,
+                _ => todo!(),
+            };
+
+            let account_validation_addr = match chain {
+                Chain::Devnet => BRIDGE_ACCOUNT_DEVNET_ETH_ADDR,
+                Chain::Holesky => BRIDGE_ACCOUNT_HOLESKY_ETH_ADDR,
+                _ => todo!(),
+            };
+
+            let contract = SudokuValidity::deploy(
+                &provider,
+                Address::from_str(state_settlement_addr).unwrap(),
+                Address::from_str(account_validation_addr).unwrap(),
+            )
+            .await
+            .map_err(|err| err.to_string())
+            .unwrap_or_else(|err| {
+                error!("{}", err);
+                process::exit(1);
+            });
+
+            info!(
+                "SudokuValidity contract successfuly deployed with address {}",
+                contract.address()
+            );
+        }
+        Command::ValidateSolution => {
+            // We could check if the specific block containing the tx is already verified, before
+            // updating the bridge's chain.
+            // let is_state_verified = is_state_verified(&state_hash, &chain, &eth_rpc_url)
+            //     .await
+            //     .unwrap_or_else(|err| {
+            //         error!("{}", err);
+            //         process::exit(1);
+            //     });
+
+            // if !is_state_verified {
+            //     info!("State that includes the zkApp tx isn't verified. Bridging latest chain...");
```

Comment:
Is this commented for some reason?

Reply:
Adding these lines would add complexity to the example as we should retrieve the state hash that includes both transactions, and send it to the Rust program somehow. So the example always updates the bridge chain instead of only updating if the state of interest isn't verified. The commented lines are for the first case, there's a small comment above the lines you mentioned also